packages/apollo-server-core/src/formatters.ts

```diff
@@ -0,0 +1,29 @@
+import { GraphQLExtension, GraphQLResponse } from 'graphql-extensions';
+import { formatApolloErrors } from 'apollo-server-errors';
+
+export class FormatErrorExtension extends GraphQLExtension {
+  private formatError: Function;
+  private debug: boolean;
+
+  public constructor(formatError: Function, debug: boolean = false) {
+    super();
+    this.formatError = formatError;
+    this.debug = debug;
+  }
+
+  public willSendResponse(o: {
+    graphqlResponse: GraphQLResponse;
+  }): void | { graphqlResponse: GraphQLResponse } {
+    if (o.graphqlResponse.errors) {
+      return {
```

Comment:
Hmmm, I'm wondering if modifying the `GraphQLResponse` passed in instead of returning a new one wouldn't be clearer as an API for extensions.

Reply:
That's totally an option, though I feel that this API is more self documenting. It's still possible to modify the `graphqlResponse` in-place too